apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -313,7 +304,51 @@ def _find_modified_include_path(self, project_dir, file_path, include_path):
         # It's probably a standard C/C++ header
         return include_path
 
+    def _cmsis_required(self, project_path: Union[str, pathlib.Path]) -> bool:
+        """Check if CMSIS dependency is required."""
+        project_path = pathlib.Path(project_path)
+        for path in (project_path / "src" / "model").iterdir():
+            if path.is_file():
+                with open(path, "r", encoding="ISO-8859-1") as lib_f:
```

Comment:
Can we do this?
```suggestion
                with open(path, "r") as lib_f:
```
If that doesn't work, let's add a comment about why.